domain/application/service/unit.go

```diff
@@ -175,18 +175,20 @@ type UnitState interface {
 	GetUnitNetNodesByName(ctx context.Context, name coreunit.Name) ([]string, error)
 }
 
-func (s *Service) makeIAASUnitArgs(units []AddUnitArg, constraints constraints.Constraints) ([]application.AddUnitArg, error) {
-	args := make([]application.AddUnitArg, len(units))
+func (s *Service) makeIAASUnitArgs(units []AddUnitArg, constraints constraints.Constraints) ([]application.AddIAASUnitArg, error) {
+	args := make([]application.AddIAASUnitArg, len(units))
 	for i, u := range units {
 		placement, err := deployment.ParsePlacement(u.Placement)
 		if err != nil {
 			return nil, errors.Errorf("invalid placement: %w", err)
 		}
 
-		arg := application.AddUnitArg{
-			Constraints:   constraints,
-			Placement:     placement,
-			UnitStatusArg: s.makeIAASUnitStatusArgs(),
+		arg := application.AddIAASUnitArg{
+			AddUnitArg: application.AddUnitArg{
+				Constraints:   constraints,
+				Placement:     placement,
+				UnitStatusArg: s.makeIAASUnitStatusArgs(),
+			},
```

Comment:
nit: I think it would make more sense to wire up `AddIAASApplication` to accept `AddIAASUnitArg` and just ignore the new `Platform` attribute for now